redis/deprecated/redisbp/monitored_client.go

```diff
@@ -30,12 +37,47 @@ func NewMonitoredFailoverClient(name string, opt *redis.FailoverOptions) *redis.
 	return client
 }
 
+// ClusterClient extends redis cluster client with a functional Wait function
+type ClusterClient struct {
+	*redis.ClusterClient
+}
+
+// WaitArgs enclose inputs for Wait command into a struct
+type WaitArgs struct {
+	Key         string
+	NumReplicas int
+	Timeout     time.Duration
+}
+
+// Wait blocks the current client until all the previous write commands are
+// successfully transferred and acknowledged by at least the specified number of replicas.
+func (cc *ClusterClient) Wait(ctx context.Context, args WaitArgs) (replicas int64, err error) {
+	if args.NumReplicas <= 0 {
+		return 0, nil
+	}
+
+	client, err := cc.ClusterClient.MasterForKey(ctx, args.Key)
+	if err != nil {
+		return 0, fmt.Errorf("redisbp: error while trying to retrieve master from key, %w", err)
+	}
+
+	replicas, err = client.Wait(ctx, args.NumReplicas, args.Timeout).Result()
+	if err != nil {
+		return 0, fmt.Errorf("redisbp: error while trying to apply replication factor, %w", err)
+	} else if int(replicas) < args.NumReplicas {
```

Comment:
🔕  hmm I'm not sure why none of the linters caught this, since you already did return in the other branch you shouldn't need to use `else` here:

```suggestion
	}
	if int(replicas) < args.NumReplicas {
```